File: frontend/services/medication.ts
import api from './api';

export interface Medication {
  id: string;
  name: string;
  dose: string;
  frequency: string;
  times: string;
  notes: string | null;
  taken_times: string | null;
  skipped_times: string | null;
  taken_today: boolean;
  created_at: string;
}

export interface Appointment {
  id: string;
  title: string;
  hospital: string;
  appointment_type: string | null;
  date: string;
  notes: string | null;
  created_at: string;
}

export const medicationService = {
  async list(): Promise<Medication[]> {
    const res = await api.get('/medications');
    return res.data;
  },

  async create(data: { name: string; dose: string; frequency: string; times: string; notes?: string }): Promise<Medication> {
    const res = await api.post('/medications', data);
    return res.data;
  },

  async update(id: string, data: Partial<Medication>): Promise<Medication> {
    const res = await api.put(`/medications/${id}`, data);
    return res.data;
  },

  async delete(id: string): Promise<void> {
    await api.delete(`/medications/${id}`);
  },

  async markTaken(id: string, time: string): Promise<Medication> {
    const res = await api.post(`/medications/${id}/taken?time=${encodeURIComponent(time)}`);
    return res.data;
  },

  async markSkip(id: string, time: string): Promise<Medication> {
    const res = await api.post(`/medications/${id}/skip?time=${encodeURIComponent(time)}`);
    return res.data;
  },

  async listAppointments(): Promise<Appointment[]> {
    const res = await api.get('/appointments');
    return res.data;
  },

  async createAppointment(data: { title: string; hospital: string; appointment_type?: string; date: string; notes?: string }): Promise<Appointment> {
    const res = await api.post('/appointments', data);
    return res.data;
  },

  async updateAppointment(id: string, data: Partial<Appointment>): Promise<Appointment> {
    const res = await api.put(`/appointments/${id}`, data);
    return res.data;
  },

  async deleteAppointment(id: string): Promise<void> {
    await api.delete(`/appointments/${id}`);
  },
};

File: backend/app/schemas/medication.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
import uuid


class MedicationCreate(BaseModel):
    name: str
    dose: str
    frequency: str
    times: str
    notes: Optional[str] = None


class MedicationUpdate(BaseModel):
    name: Optional[str] = None
    dose: Optional[str] = None
    frequency: Optional[str] = None
    times: Optional[str] = None
    notes: Optional[str] = None


class MedicationResponse(BaseModel):
    id: uuid.UUID
    name: str
    dose: str
    frequency: str
    times: str
    notes: Optional[str]
    taken_times: Optional[str] = None
    skipped_times: Optional[str] = None
    taken_today: bool
    created_at: datetime

    class Config:
        from_attributes = True


class AppointmentCreate(BaseModel):
    title: str
    hospital: str
    appointment_type: Optional[str] = None
    date: datetime
    notes: Optional[str] = None


class AppointmentUpdate(BaseModel):
    title: Optional[str] = None
    hospital: Optional[str] = None
    appointment_type: Optional[str] = None
    date: Optional[datetime] = None
    notes: Optional[str] = None


class AppointmentResponse(BaseModel):
    id: uuid.UUID
    title: str
    hospital: str
    appointment_type: Optional[str]
    date: datetime
    notes: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True
